bahmnicore-omod/src/test/java/org/bahmni/module/bahmnicore/web/v1_0/controller/VisitDocumentControllerTest.java

```diff
@@ -160,4 +160,35 @@ public void shouldNotCallDeleteWithGivenFileNameIfFileNameIsEmpty() throws Excep
         ResponseEntity<Object> responseEntity = visitDocumentController.deleteDocument("");
         Assert.assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
     }
+
+    @Test
+    public void shouldFailIfFileNameWithSpecialCharsOtherThanDashAndUnderscoreIsPassedInRequest() throws Exception {
+        PowerMockito.mockStatic(Context.class);
+        PowerMockito.when(Context.getPatientService()).thenReturn(patientService);
+        Patient patient = new Patient();
+        patient.setId(1);
+        patient.setUuid("patient-uuid");
+        when(patientService.getPatientByUuid("patient-uuid")).thenReturn(patient);
+
+        Document document = new Document("abcd", "jpeg", "consultation", "patient-uuid", "image", "file/name");
+
+        visitDocumentController.saveDocument(document);
```

Comment:
What are we asserting here?
If you expect an exception to be thrown you need to assert on that exception
`@Test(expected = InvalidInputException.class)`

Reply:
@shilpa-iplit - please provide a comment when you resolve, to say how/what you did to resolve this comment.

Reply:
Added throws Exception in test function so that exception is verified if filename contains invalid characters.

Reply:
![AllPatientDocument](https://user-images.githubusercontent.com/38578773/188499301-f2e01f3e-4957-4a37-873e-f9d5a2f4cee4.png)
<img width="425" alt="patientDocument" src="https://user-images.githubusercontent.com/38578773/188499308-e0d888e5-26d9-42d4-94f6-bca022812a05.png">

Old files will follow: patientid-encounterName-uuid.ext (eg. 6-Patient-Document-706a448b-3f10-11e4-adec-0800271c1b75.png)
New ones will follow: patientid_encounterName_uuid__filename.ext (eg. 6-Patient-Document-706a448b-3f10-11e4-adec-0800271c1b75__doc1.png)
Please review  @gsluthra   @N0-man